src/components/InvitationForm/sagas.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  takeLatest,
+  put,
+  call,
+} from 'redux-saga/effects'
+import request from 'util/request'
+import {
+  InvitationFormActionsTypes,
+  acceptInvitationWithSignInActions,
+  acceptInvitationActions,
+  rejectInvitationActions,
+  validateInvitationTokenActions,
+  invitationSignInActions,
+} from './ducks'
+import { openNotification } from 'containers/NotificationStack/ducks'
+import { API_URL } from 'constants/endpoints'
+import {
+  InvitationResponse,
+} from './types'
+import stateGenerator from 'util/stateGenerator'
+import { authActions } from 'containers/Auth/ducks'
+
+const STATE_STORAGE = 'ssoStateStorage'
+const STATE_STORAGE_INVITATION = 'ssoStateInvitationStorage'
+
+export function * invitationWithSignInSaga (
+  action: ReturnType<typeof acceptInvitationWithSignInActions.request>,
+) {
+  try {
+    yield call(request, {
+      url: `${API_URL}/auth/oidc/${action.payload.provider}/token?invite=true`,
+      method: 'POST',
+      data: { code: action.payload.code },
+    })
+
+    yield call(request, {
+      url: `${API_URL}/invites/${action.payload.token}/accept`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+
+    localStorage.removeItem(STATE_STORAGE)
+    localStorage.removeItem(STATE_STORAGE_INVITATION)
+    yield put(authActions.loginSuccess())
+    yield put(openNotification('success', 'You have accepted your invitation.', 4000))
+    window.location.href = window.location.origin
```

Comment:
Don’t manipulate history directly as it may cause side effects. Use history in the middleware. You can trigger this in a middleware by calling an action.